src/main/java/net/william278/velocitab/player/TabPlayer.java

```diff
@@ -126,18 +126,10 @@ public CompletableFuture<String> getNametag(@NotNull Velocitab plugin) {
 
     @NotNull
     public CompletableFuture<String> getTeamName(@NotNull Velocitab plugin) {
-        if (!plugin.getSettings().isSortPlayers()) {
-            return CompletableFuture.completedFuture("");
-        }
-
-        final String sortingFormat = String.join("", plugin.getSettings().getSortingElements());
-        return Placeholder.replace(sortingFormat, plugin, this) // Replace placeholders
-                .thenApply(formatted -> formatted.length() > 12 ? formatted.substring(0, 12) : formatted) // Truncate
-                .thenApply(truncated -> truncated + getPlayer().getUniqueId().toString().substring(0, 4)) // Make unique
-                .thenApply(teamName -> this.teamName = teamName)
-                .exceptionally(e -> {
-                    plugin.log(Level.ERROR, "Failed to get team name for " + player.getUsername(), e);
-                    return "";
+        return plugin.getSortingManager().getTeamName(this)
+                .thenApply(teamName -> {
+                    this.teamName = teamName;
+                    return teamName;
                 });
```

Comment:
```suggestion
        return plugin.getSortingManager().getTeamName(this)
                .thenApply(teamName -> {
                    return this.teamName = teamName;
                });
```

Use Optional#map.orElse if you make getSortingManager return an Optional